### ISSUE TITLE
Whitelist capitalized error strings if they are used in a method and begin with the receiver type name.

### ISSUE THREAD

**Issue by mewmew** [golang/lint#36]
Some error messages use the following prefix `T.M: error message` where `T` refers to the receiver type of a method `M`. Below are some examples from the standard library:

`math/big/int.go:507:21: error strings should not be capitalized`

``` go
func (z *Int) Scan(s fmt.ScanState, ch rune) error {
...
        return errors.New("Int.Scan: invalid verb")
```

`math/big/rat.go:414:21: error strings should not be capitalized`

``` go
func (z *Rat) Scan(s fmt.ScanState, ch rune) error {
...
        return errors.New("Rat.Scan: invalid verb")
```

`time/time.go:853:27: error strings should not be capitalized`

``` go
func (t Time) MarshalBinary() ([]byte, error) {
...
            return nil, errors.New("Time.MarshalBinary: zone offset has fractional minute")
```

What are your thoughts on whitelisting this specific use of capitalized error strings?
